### PR TITLE
Add CONTRIBUTION guidelines and new README

### PR DESCRIPTION
**Pull request by jvican**
This commit adds more complete information to the CONTRIBUTING and
README file to be more friendly towards contributors.

I think this is super important to attract future contributors and I've
tried to be as concise as possible. More changes to this section will
come in the future.

### CODE REVIEW

**Comment by jvican** [comment]
These changes are necessary for some tickets in the issue tracker that I will propose in the Scala Center hackathon taking place on this Friday at 17:00, at Scalasphere 2017.

**Comment by jvican** [comment]
@dwijnand changes have been merged.

**Review by dwijnand — APPROVED** [review]

**Comment by jvican** [comment]
@eed3si9n We need this for today's hackathon, can you have a look?

File: README.md [hunk 1]
```diff
@@ -1,21 +1,61 @@
 Zinc
 ====
 
-Zinc is an incremental compilation module of sbt with the sbt-agnostic API.
+Zinc is the incremental compiler for Scala. Most Scala developers use it daily
+without noticing -- it's embedded in key build tools like sbt, CBT and pants.
```
Comment (eed3si9n) [inline README.md:5]: The old republished "Zinc" *was* embedded into sbt, etc. the new Zinc is a standalone incremental compiler library intending to be used by everyone. sbt 1 *will* use it, and I think Pants is committed to using it, but I don't know of others yet.

**Comment by eed3si9n** [comment]
I have a minor comment, but overall looks really good.

File: README.md (hunk 1 above)
Comment (jvican) [reply README.md:5]: Pants is using it, and I know that CBT is interested in using it as well. Bazel will not.

Comment (jvican) [reply README.md:5]: I'm updating the document with this information.

**Comment by jvican** [comment]
This PR has been updated with your suggestion @eed3si9n. Some of it was already described in the "History" section of the README, and now it's been enriched with more information.

**Review by eed3si9n — APPROVED** [review]

**Comment by jvican** [comment]
I got two signoff's, merging.

**Pull request merged**